packages/validation/src/permissions/research-output.ts

```diff
@@ -2,9 +2,42 @@ import { User } from '@asap-hub/auth';
 import { UserRole, UserResponse } from '@asap-hub/model';
 import { isEnabled } from '@asap-hub/flags';
 
+type user = Omit<User, 'algoliaApiKey'> | UserResponse;
+type association = 'teams' | 'workingGroups';
+
+export const isUserProjectManager = (
+  user: user | null,
+  association: association,
+  associationIds: string[],
+): boolean => {
+  if (user === null) {
+    return false;
+  }
+  return user[association].some(
+    (teamOrWorkingGroup) =>
+      associationIds?.includes(teamOrWorkingGroup.id) &&
+      teamOrWorkingGroup.role === 'Project Manager',
+  );
+};
+
+export const isUserMember = (
+  user: user | null,
+  association: association,
+  associationIds: string[],
+): boolean => {
+  if (user === null) {
+    return false;
+  }
+  return user[association].some(
+    (teamOrWorkingGroup) =>
+      associationIds?.includes(teamOrWorkingGroup.id) &&
+      teamOrWorkingGroup.role !== 'Project Manager',
+  );
+};
```

Comment:
To be honest, I'm not a big fan of this refactor because we're not using `isUserProjectManager` separated from asap role = staff anywhere now and this might make it easier to use only `isUserProjectManager` and lead to error in the permission.

Also, I think it's a little misleading having `teamOrWorkingGroup.role !== 'Project Manager'` in `isUserMember` because a PM is still a member of the team/wg, but he/she is also a PM 😝 

I would remove `isUserProjectManager` and `isUserMember` and leave `getUserRole` the same as before

Reply:
Ok I will drop the changes done here